### ISSUE TITLE
Allow Swift in code base?

### ISSUE THREAD

**Issue by danielpunkass** [iMediaSandboxing/iMedia#109]
I'm looking again at implementing NSCollectionView support. What are thoughts about whether the iMedia project should start implementing new code in Swift instead of Objective-C?

**Comment by danielpunkass** [comment]
I guess the question has problem been complicated by lack of ABI stability in Swift. Since I compile iMedia to suit my app, it is not an issue for me. But if any clients are still dependent on binary deployments of iMedia.framework, I guess that's a consideration.

**Comment by iljaiwas** [comment]
Though call. While we work on some Swift apps, the app we use iMedia for is still Objective-C only. Not sure if we'd want to ship the Swift runtime only to support iMedia.
On the other hand, the transition has to start at same point. 🤔
What are the @jjac  and @gloubibou thinking.

**Comment by gloubibou** [comment]
I vote against mixing Swift and Objective-C in the project. For an all-new project, I would consider Swift even though I am still biased in favor of Objective-C. For an existing application, I would not consider the additional overhead of managing two languages in one project or shipping the Swift runtime.

For an open-source framework I am even less likely to consider a partial migration to Swift. You would create a version of iMedia browser that as a slightly better grid view. For this host applications would need to ship the Swift runtime. I would bet on people creating forks from the Obj-C only version.
 
@danielpunkass I have recently worked to replace IKImageBrowserView with NSCollectionView in HoudahSpot 5.0 (still in beta). It took some effort to replicate all desirable features of IKImageBrowserView. Let me know if I can help with NSCollectionView in iMedia.

**Comment by danielpunkass** [comment]
Sounds like there is a good enough reason to stick with all ObjC for now. @gloubibou that would be great if you can share your progress on the IKImageBrowserView replacement. I started hacking away at it here: https://github.com/danielpunkass/iMedia/tree/collection-view-revamp, but there are still a lot of rough edges. Particuarly if you've already tackled the skimmability stuff that would be handy to have!

If you are prepared to share your changes on a public branch I would probably just integrate them and then propose we consider merging them to iMediaSandbox/develop at some point.

**Comment by danielpunkass** [comment]
Let's stick to all ObjC for now.

**Comment by gloubibou** [comment]
@danielpunkass I did not work on a IKImageBrowserView replacement for iMedia. I replaced an instance of IKImageBrowserView with a NSCollectionView in a different application. There however were things I needed to rebuild that could be useful to iMedia. From the top of my head: lazy loading Quick Look, zoomable flow layout, maximum zoom that tracks view size, selection highlight, arrow key navigation, … 

I also wanted "shift-click range selections". That required private API.

Pierre
